### ISSUE TITLE
Light/dark mode leads to low background/colour contrast tooltip

### ISSUE THREAD

**Issue by mcanouil** [quarto-dev/quarto-cli#13195]
Currently the `$tooltip-bg` and `$tooltip-colour` have no contrast or insufficient one in dark mode.

The main case is quarto.org, see:

- https://quarto.org/docs/authoring/code-annotation.html

**Comment by cscheid** [comment]
Thank yoU!

**Comment by mcanouil** [comment]
I've added a more general reprex with brand showing that the tooltip is always wrong in dark mode (dark background)

I've personally redefined the variables as:

```
$tooltip-color: $background !default;
$tooltip-bg: $foreground !default;
```

**Comment by xx02al** [comment]
Hey @mcanouil thanks for inviting me to take a look at this issue, so I do. Honestly, unfortunately I don't understand the point you're trying to make.

I've tweaked the reprex a bit, especially to also include some tooltips. (That's the docs [code annotation example](https://quarto.org/docs/authoring/code-annotation.html#annotation-syntax)). I kept the Cosmo theme as the theme to inherit from in dark mode, because you set it, but just to have reminded: its not a dark mode like theme (but that doesn't stop us from using it as a "preset", so we can do that). Furthermore, I did set a different light theme because I found it illustrates more clear than when we have the same colors in dark and light mode.

```` qmd
---
format: html
theme:
  light: flatly
  dark:
    - cosmo
    - theme-dark.scss
code-annotations: hover
---

This is some text in the `$body-color` color.

```r
library(tidyverse)
library(palmerpenguins)
penguins |>                                      # <1>
  mutate(                                        # <2>
    bill_ratio = bill_depth_mm / bill_length_mm, # <2>
    bill_area  = bill_depth_mm * bill_length_mm  # <2>
  )                                              # <2>
```
1. Take `penguins`, and then,
2. add new columns for the bill ratio and bill area.

This is some text in the `$body-color` color.
````

`theme-dark.scss` is then:

```
/*-- scss:defaults --*/

$deepindigo: #11063e;
$lightpink: #fdf7ff;

$body-bg: $deepindigo;
$body-color: $lightpink;

```

With `theme-dark.scss` we inherit from `cosmo` and then from the default Bootstrap theme.

For me it looks correct and also the tooltip text-background has good contrast values. Also if we would replace our current dark with a more dark mode like [Bootswatch](https://bootswatch.com/) theme, Cyborg or Darkly for example.

**Comment by mcanouil** [comment]
The point was not: "if you tweak everything then it's correct thus users should just tweak everything".
The point was that since Quarto allows any theme to be used as dark theme and sets defaults sometimes different from the original ones, changing background or foreground via brand should not require users to have to create a whole theme.

And that's exactly the same point for the other issue.

**Comment by xx02al** [comment]
Well, if we want to leave the current example aside, then here is my opinion on the more fundamental point.

I think it would be reasonable for Quarto to make some defaults (tweaks) when a user *uses brand* for theming in cases where themes have been combined "unhappily". Brand is something of a lightweight/gentle entry point, and as such, such tweaks would be a reasonable expectation from users. When theming is done *without brand*, however, I find the inheritance principle strong as it is.

**Comment by mcanouil** [comment]
It's not two different systems, it's two layers.

Changing the defaults, for example, to be conditional on the background or foreground will change the behaviour with and without brand.

**Comment by cwickham** [comment]
@mcanouil I'm trying to understand and reproduce this problem. Piecing together from above I have:
````qmd
---
format: html
theme:
  dark:
    - cosmo
    - brand
brand: 
  color:
    background: "#333333"
    foreground: "#fafafa"
code-annotations: hover
---

```r
library(tidyverse)
library(palmerpenguins)
penguins |>                                      # <1>
  mutate(                                         # <2>
    bill_ratio = bill_depth_mm / bill_length_mm,  # <2>
    bill_area  = bill_depth_mm * bill_length_mm   # <2>
  )                                               # <2>
```
1. Take `penguins`, and then,
2. add new columns for the bill ratio and bill area.
````

Which gives me tooltips with the same background and foreground as the page (Quarto 1.10.12):

<img width="883" height="203" alt="Image" src="https://github.com/user-attachments/assets/fa5ad485-4ea2-43fc-8c48-f5a7b5a2d461" />

The contrast between text and background seems sufficient. But, maybe this was about the constrast between the tooltip and the page? I.e. in a wider viewport the tooltip floats right and then its only distinction is a faint border. Any clarification you can provide would be helpful. Or maybe there has been a change in brand or dark mode that fixed this?

**Comment by mcanouil** [comment]
I need to go back into it and behaviour might have change.
What didn't and is related to the other theme issue where Quarto CLI hardcoded default colours/background for code because the custom style was lacking some.

For example, if you hover the first annotation of your example, you won't be able to actually see the code from the annotation. (the ugly highlight rectangle shift towards left is reported also in another issue, it's because of the DOM change for the copy button).

(I believe this issue was created as a follow-up of a slack discussion with Carlos, thus his "thank you")

**Comment by mcanouil** [comment]
It might have been about the "copied" tooltip appearing on copy:

<img width="181" height="65" alt="copied tooltip" src="https://github.com/user-attachments/assets/80c1f64e-cdfb-461a-9581-0ff5768a28ab" />

This can also be seen on quarto.org in dark mode.
